lib/api/core/validation/index.js

```diff
@@ -1,690 +1,680 @@
-
-  this.rawConfiguration = {};
-}
+    /** @type {Dsl} */
+    this.dsl = new Dsl();
 
-/**
- * Walks through all types in "defaultTypesFiles" initializes all types
- */
-Validation.prototype.init = function validationInit () {
-  var ValidationType;
-
-  [
-    'anything',
-    'boolean',
-    'date',
-    'email',
-    'enum',
-    'geoPoint',
-    'geoShape',
-    'integer',
-    'ipAddress',
-    'numeric',
-    'object',
-    'string',
-    'url'
-  ].forEach(typeFile => {
-    ValidationType = require(`./types/${typeFile}`);
-    this.addType(new ValidationType());
-  });
-};
+    this.rawConfiguration = {};
+  }
 
-/**
- * @param {Request} request
- * @param {boolean} verbose
- * @returns {Promise.<{documentBody: *, errorMessages:string[], valid: boolean}|KuzzleRequest>}
- */
-Validation.prototype.validationPromise = function validationValidationPromise (request, verbose) {
-  var
-    id = request.input.resource._id,
-    updateBodyClone,
-    isUpdate = false,
-    collectionSpec = {};
-
-  verbose = typeof verbose !== 'undefined' && verbose;
-
-  if (this.specification && this.specification[request.input.resource.index]
-    && this.specification[request.input.resource.index][request.input.resource.collection]) {
-    collectionSpec = this.specification[request.input.resource.index][request.input.resource.collection];
+  /**
+   * Walks through all types in "defaultTypesFiles" initializes all types
+   */
+  init() {
+    [
+      'anything',
+      'boolean',
+      'date',
+      'email',
+      'enum',
+      'geoPoint',
+      'geoShape',
+      'integer',
+      'ipAddress',
+      'numeric',
+      'object',
+      'string',
+      'url'
+    ].forEach(typeFile => {
+      const TypeConstructor = require(`./types/${typeFile}`);
+      this.addType(new TypeConstructor());
+    });
   }
 
-  return new Promise(resolve => {
-    var updateRequest;
-
-    if (request.input.controller === 'write' && request.input.action === 'update') {
-      isUpdate = true;
-      updateRequest = new Request({
-        index: request.input.resource.index,
-        collection: request.input.resource.collection,
-        controller: 'read',
-        action: 'get',
-        _id: id
-      });
-      this.kuzzle.services.list.storageEngine.get(updateRequest).then(document => {
-        // Avoid side effects on the request during the update validation
-        updateBodyClone = _.cloneDeep(request.input.body);
+  /**
+   * @param {Request} request
+   * @param {boolean} [verbose]
+   * @returns {Promise.<{documentBody: *, errorMessages:string[], valid: boolean}|KuzzleRequest>}
+   */
+  validationPromise(request, verbose = false) {
+    const
+      id = request.input.resource._id;
+    let
+      collectionSpec = {},
+      updateBodyClone,
+      isUpdate = false;
+
+    if (this.specification && this.specification[request.input.resource.index]
+      && this.specification[request.input.resource.index][request.input.resource.collection]) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 95 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)